WordPress/src/main/java/org/wordpress/android/ui/media/MediaBrowserActivity.java

```diff
@@ -665,36 +665,39 @@ public void onSupportActionModeFinished(@NonNull ActionMode mode) {
     }
 
     public void deleteMedia(final ArrayList<Integer> ids) {
-        Set<String> sanitizedIds = new HashSet<>(ids.size());
-
         final ArrayList<MediaModel> mediaToDelete = new ArrayList<>();
+        int processedItemCount = 0;
         // Make sure there are no media in "uploading"
         for (int currentId : ids) {
             MediaModel mediaModel = mMediaStore.getMediaWithLocalId(currentId);
             if (mediaModel == null) {
                 continue;
             }
 
-            if (WPMediaUtils.canDeleteMedia(mediaModel)) {
+            // only able to delete items that are not being uploaded for a post
+            // i.e. standalone uploading items can be deleted, but items inserted in a Post can't
+            if (!UploadService.isMediaBeingUploadedForAPost(mediaModel)) {
+
+                // if uploading, first issue a cancel upload command
+                if (UploadService.isPendingOrInProgressMediaUpload(mediaModel)) {
+                    MediaStore.CancelMediaPayload payload = new MediaStore.CancelMediaPayload(mSite, mediaModel, false);
+                    mDispatcher.dispatch(MediaActionBuilder.newCancelMediaUploadAction(payload));
+                }
+
                 if (mediaModel.getUploadState() != null &&
                         MediaUtils.isLocalFile(mediaModel.getUploadState().toLowerCase())) {
```

Comment:
well it actually could be null, and it needs to be made sure to be lowercase so the implementation of `isLocalFile()` can work correctly. We might refactor this and convert to `MediaUploadState` properly first, using `MediaUploadState.fromString()`, and making `isLocalFile()` receive a `MediaUploadState` parameter type instead of a String, but maybe in another smallish PR as this is also being used like that on some other parts of the code? will do that right after this one 👍